x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupShardStatus.java

```diff
@@ -68,15 +108,57 @@ public RollupShardStatus(StreamInput in) throws IOException {
         numSent = in.readLong();
         numIndexed = in.readLong();
         numFailed = in.readLong();
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_017) && in.readBoolean()) {
+            totalDocCount = in.readLong();
+            totalShardDocCount = in.readLong();
+            lastSourceTimestamp = in.readLong();
+            lastTargetTimestamp = in.readLong();
+            lastIndexingTimestamp = in.readLong();
+            rollupBeforeBulkInfo = in.readNamedWriteable(RollupBeforeBulkInfo.class, RollupBeforeBulkInfo.NAME);
```

Comment:
Why named writeable and not just writeable here? I think we know what class we need to deserialise here?

Reply:
Just use: `new RollupBeforeBulkInfo(in)`?